source/validate_id.cpp

```diff
@@ -2377,6 +2382,29 @@ function<bool(unsigned)> getCanBeForwardDeclaredFunction(SpvOp opcode) {
 
 namespace libspirv {
 
+spv_result_t UpdateIdUse(ValidationState_t& _) {
+  for (const auto& inst : _.ordered_instructions()) {
+    for (auto& operand : inst.operands()) {
+      const spv_operand_type_t& type = operand.type;
+      const uint32_t operand_id = inst.words()[operand.offset];
+
+      switch (type) {
+        case SPV_OPERAND_TYPE_VARIABLE_ID:
```

Comment:
Good idea. I am skipping SPV_OPERAND_TYPE_RESULT_ID in this switch but I can add a continue condition for that one.